algebra/src/main/java/com/hubspot/algebra/Result.java

```diff
@@ -44,6 +44,11 @@ public void ifErr(Consumer<? super ERROR_TYPE> consumer) {
     Results.getErr(this).ifPresent(consumer);
   }
 
+  public void ifOkOrErr(Consumer<? super SUCCESS_TYPE> okConsumer, Consumer<? super ERROR_TYPE> errConsumer) {
```

Comment:
I wonder if we should call this `match` since it has a similar purpose to the existing `match` method but is meant for side effects instead of a return value.